app/publishing/calculation/get-fundings-by-calculation-id.js

```diff
@@ -0,0 +1,16 @@
+const db = require('../../data')
+
+const getFundingsByCalculationId = async (calculationId, transaction) => {
+  return db.funding.findAll({
+    lock: true,
+    skipLocked: true,
+    where: {
+      calculationId
+    },
+    attributes: ['calculationId', 'fundingCode', 'areaClaimed', 'rate'],
+    raw: true,
+    transaction
+  })
+}
+
+module.exports = getFundingsByCalculationId
```

Comment:
We are not making any update to funding table after/while being read, I do not think we need a lock here. locking block of records simply means other process can not access it, if we have multiple locks and they are not properly managed, it may result in deadlock.

Reply:
I originally did not have a lock here because of the database relationship which exists of 1:N between calculation and funding and locking calculation is enough to transitivity lock funding because

if fundingA and fundingB belongs to calculationA
and funding C belongs to calculationB
then when processA picks up calculationA the only fundings it is going to look at are fundingA and fundingB, i.e. it will not look at fundingC or any other because they do not link to calculationA

However, the reason I added it was 

if processA picked up calculationA
and somehow processB picked up calculationA
then it would be a race condition as to pick process picks up fundingA and fundingB
which locks it for the other, skips over, returns no funding and fails at validation
other process will complete and mark the message as complete

Not sure how deadlock can occur in this due to the referential integrity of the db setup.

Reply:
```
if processA picked up calculationA
and somehow processB picked up calculationA
```
The following position should not happen as we already have lock on calculation.
Should it happen, then there would be high probability for deadlock as follows:

Process A with calculation A may be processing funding A, hence funding A would be locked to process A
Process B with calculation A may be processing funding B, hence funding B would be locked to process B

Process A would be waiting for funding B to be released by Process B for it (Process A) to proceed, at the same time, Process B would be waiting for funding A to be released by Process A for it (Process B) to proceed, at this point there is deadlock between Process A and Process B.

Reply:
Have removed.